tabled/src/settings/split/mod.rs

```diff
@@ -0,0 +1,333 @@
+
+    /// Returns a split [`Table`] with all cells retained.
+    ///
+    /// ```text
+    ///                                                 +---+---+---+
+    ///                                                 | a | b | c |
+    ///                                                 +---+---+---+
+    /// +---+---+---+---+---+                           | d | e |   |
+    /// | a | b | c | d | e | Split::column(3).retain() +---+---+---+
+    /// +---+---+---+---+---+           =>              | f | g | h |
+    /// | f | g | h |   |   |                           +---+---+---+
+    /// +---+---+---+---+---+             |-----------> |   |   |   |
+    ///               ^   ^               |             +---+---+---+
+    ///               |___|_____cells are kept!
+    /// ```
+    ///
+    /// ## Notes
+    ///
+    /// See [`clean`] for an alternative display option.
+    ///
+    /// [`Table`]: crate::Table
+    pub fn retain(self) -> Self {
+        Self {
+            display: Display::Retain,
+            ..self
+        }
+    }
+}
+
+impl<R, D, Cfg> TableOption<R, D, Cfg> for Split
+where
+    R: Records + ExactRecords + Resizable,
+{
+    fn change(&mut self, records: &mut R, _: &mut Cfg, _: &mut D) {
+        // prepare many variables
+        let Split {
+            direction,
+            behavior,
+            display,
+            index: section_length,
+        } = *self;
+        let columns = records.count_columns();
+        let rows = records.count_rows();
+        if columns == 0 || rows == 0 || section_length == 0 {
+            return;
+        }
+        let (primary_length, secondary_length) = match direction {
+            Direction::Column => (columns, rows),
+            Direction::Row => (rows, columns),
+        };
+        let sections_per_direction = ceil_div(primary_length, section_length);
+        let mut filtered_sections = 0;
+
+        // determine section over direction or vice versa based on behavior
+        let (outer_range, inner_range) = match behavior {
+            Behavior::Concat => (1..sections_per_direction, 0..secondary_length),
+            Behavior::Zip => (0..secondary_length, 1..sections_per_direction),
+        };
+        for outer_index in outer_range {
+            for inner_index in inner_range.clone() {
+                let (section_index, from_secondary_index, to_secondary_index) = match behavior {
+                    Behavior::Concat => (
+                        outer_index,
+                        inner_index,
+                        inner_index + outer_index * secondary_length - filtered_sections,
+                    ),
+                    Behavior::Zip => (
+                        inner_index,
+                        usize::checked_sub(outer_index * sections_per_direction, filtered_sections)
+                            .unwrap_or(0),
+                        outer_index * sections_per_direction + inner_index - filtered_sections,
+                    ),
+                };
```

Comment:
maybe make it a function?

Reply:
I know you already flagged this as resolved, but I followed your advice and I think it's cleaner now 😄